### PR TITLE
Update xcode to 14c18 for host only bots

### PR DESCRIPTION
**Pull request by keyonghan**
With https://flutter-review.googlesource.com/c/recipes/+/42160, we are ready to make all tests consistent with xcode 14c18.

### CODE REVIEW

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:2514]: Instead of setting this at the builder level can we just set it once at the top `mac_benchmark`?

Comment (keyonghan) [reply .ci.yaml:2514]: Seems `Mac_benchmark flutter_tool_startup__macos` doesn't need an xcode. I guess it is okay for it to ask for an xcode if we move to the platform level?

**Review by jmagman — APPROVED** [review]
LGTM, thank you so much @keyonghan!

**Review by jmagman — CHANGES_REQUESTED** [review]
Temporarily blocking this, @zanderso has requested we hold off on this (and the inevitable slow down populating caches) until the very delayed engine -> framework roll gets unwedged.  https://github.com/flutter/flutter/pull/125221

@zanderso can you let us know when this is safe to land?

**Comment by keyonghan** [comment]
Yeah, removed the `auto-submit` label to postpone to after hour land.

**Review by jmagman — APPROVED** [review]
Looks like the roll succeeded https://github.com/flutter/flutter/pull/125271
LGTM!

**Pull request merged**
